### PR TITLE
Update documentation build to use "assets"

### PR DESCRIPTION
**Pull request by MatthewHambley**
GitHub have changed the way they handle "pages."

In the past they were taken from a branch in the repository called "gh_pages." Now they are tarred up as an "asset" and installed from there.

This change reflects that.

I've tested it as far as I can but it's always hard to be certain if workflows are correct until they are deployed.

### CODE REVIEW

**Review by hiker — APPROVED** [review]
I have asked our git expert Scott Wales to review this, and he said it's all good, so I'll proceed to merge it.

**Comment by hiker** [comment]
@MatthewHambley , could you please fix the conflicts, then I'm happy to merge.

**Review by hiker — APPROVED** [review]
Proceeding to merge.

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
+name: Build Documentation
+
+on:
+  push:
+    branches:
+      - 'main'
```
Comment (yaswant) [inline .github/workflows/documentation.yml:6]: This workflow targets `main`, but there is not such a branch - Perhaps Master needs renaming as main?

Comment (hiker) [reply .github/workflows/documentation.yml:6]: Thanks, I've opened #321 for this.

Comment (hiker) [reply .github/workflows/documentation.yml:6]: @yaswant , thanks, fixed in 41576067353cd0cf3447b86ca5b478de8d706935

**Pull request merged**